url79f898233e_tombrus_jvthundred_tgz-pJ8-com_tombrus_vthundred_test_screen_BorderJ8/src/com/tombrus/vthundred/terminal/TerminalImpl.java

```diff
@@ -0,0 +1,358 @@
+    }
+
+    private void cursorOff () throws IOException {
+        writeControlSeq(type.getCursorOffSeq());
+    }
+
+    private void reportPosition () throws IOException {
+        effectuateRequestedPosAndProps();
+        writeControlSeq(type.getReportPosSeq());
+    }
+
+    protected void startActions () throws IOException {
+        watchDog = new Thread() {
+            @Override
+            public void run () {
+                while (watchDog ==this) {
+                    try {
+                        sleep(1000);
+                    } catch (InterruptedException e) {
+                        DB.error("watchdog hit an interrupt");
+                    }
+                    if (DB.hasErrors()) {
+                        System.exit(999);
+                    }
+                }
+            }
+        };
+        watchDog.setDaemon(true);
+        watchDog.start();
+        enterPrivateMode();
+        clearScreen();
+        write(0, 0);
+        probeTerminalSize();
+    }
+
+    protected void stopActions () throws IOException {
+        requestedCursorOn = true;
+        requestedProps    = CharProps.DEFAULT;
+        requestedX        = 0;
+        requestedY        = 0;
+        effectuateRequestedPosAndProps();
+        graphWriter.graphOff();
+        exitPrivateMode();
+        flush();
+        watchDog = null;
+        inputDecoder.setScreenSize(null);
+    }
+
+    public void addResizeHandler (ResizeHandler handler) {
+        inputDecoder.addResizeHandler(handler);
+    }
+
+    public void removeResizeHandler (ResizeHandler handler) {
+        inputDecoder.removeResizeHandler(handler);
+    }
+
+    public void addKeyHandler (KeyHandler h) {
+        inputDecoder.addKeyHandler(h);
+    }
+
+    public void removeKeyHandler (KeyHandler h) {
+        inputDecoder.removeKeyHandler(h);
+    }
+
+    protected void probeTerminalSize () {
+        try {
+            lockWithStartIfNeeded();
+            DB.t("\n<probeTerminalSize>\n");
+            final CountDownLatch latch   = new                 CountDownLatch(1);
+            final ResizeHandler  handler = new ResizeHandler() {
+                public void handleResize (TerminalXY newSize) {
+                    //DB.t("\n<latch downed>\n");
+                    latch.countDown();
+                }
+            };
+            inputDecoder.setProbeSizeHandler(handler);
+            write(5000, 5000);
+            reportPosition();
+            write(   0, 0   );
+            flush();
+            if (!isOnConsole()) {
+                inputDecoder.setScreenSize(DEFAULT_SCREEN_SIZE);
+                DB.t("\n<no console, no answer expected, defaulting to " +DEFAULT_SCREEN_SIZE +">\n");
+            }else {
+                try {
+                    if (!latch.await(1000, TimeUnit.MILLISECONDS)) {
+                        DB.error("terminal did not respond with its size in time");
+                    }
+                } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)